source/server/config/network/http_connection_manager.cc

```diff
@@ -33,6 +33,13 @@ static Registry::RegisterFactory<Singleton::RegistrationImpl<date_provider_singl
                                  Singleton::Registration>
     date_provider_singleton_registered_;
 
+static constexpr char route_config_provider_manager_singleton_name[] =
+    "route_config_provider_manager_singleton_name";
+static Registry::RegisterFactory<
+    Singleton::RegistrationImpl<route_config_provider_manager_singleton_name>,
```

Comment:
I wonder if it makes sense to have a macro to avoid the boilerplate when we declare singletons. E.g. make this something more like `DEFINE_SINGLETON(route_config_provider_manager);`, which then synthesizes the rest. Could clean this up for the date provider too.